diff --git a/haskell/haskell.bzl b/haskell/haskell.bzl
index 3ae7989..dcb31f1 100644
--- a/haskell/haskell.bzl
+++ b/haskell/haskell.bzl
@@ -236,7 +236,7 @@ haskell_library = rule(
             doc = "A dictionary mapping dependencies to module reexports that should be available for import by dependencies.",
         ),
         linkstatic = attr.bool(
-            default = False,
+            default = True,
             doc = "Create a static library, not both a static and a shared library.",
         ),
         package_name = attr.string(
diff --git a/haskell/private/actions/link.bzl b/haskell/private/actions/link.bzl
index 6298119..65ccfaf 100644
--- a/haskell/private/actions/link.bzl
+++ b/haskell/private/actions/link.bzl
@@ -354,7 +354,7 @@ def link_library_dynamic(hs, cc, dep_info, cc_info, extra_srcs, objects_dir, my_
         hs = hs,
         cc_info = cc_info,
         dynamic = False,
-        pic = True,
+        pic = False,
         binary = dynamic_library,
         args = args,
     )
diff --git a/haskell/private/haskell_impl.bzl b/haskell/private/haskell_impl.bzl
index d940c72..cb6cb59 100644
--- a/haskell/private/haskell_impl.bzl
+++ b/haskell/private/haskell_impl.bzl
@@ -147,8 +147,9 @@ def _expand_make_variables(name, ctx, strings):
         ctx.attr.plugins,
         ctx.attr.tools,
     ]
-    targets = [target for attr in label_attrs for target in attr]
-    strings = [ctx.expand_location(str, targets) for str in strings]
+    for attr in label_attrs:
+        targets = [target for target in attr]
+        strings = [ctx.expand_location(str, targets) for str in strings]
     strings = [ctx.expand_make_variables(name, str, {}) for str in strings]
     return strings
 
diff --git a/haskell/providers.bzl b/haskell/providers.bzl
index 961d4d3..7fff7fe 100644
--- a/haskell/providers.bzl
+++ b/haskell/providers.bzl
@@ -207,7 +207,7 @@ def get_ghci_extra_libs(hs, cc_info, dynamic = True, path_prefix = None):
         hs,
         cc_info,
         dynamic = dynamic,
-        pic = True,
+        pic = False,
         fixup_dir = "_ghci_libs",
     )
     libs = depset(transitive = [static_libs, dynamic_libs])
